WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductImageUploaderTests.swift

```diff
@@ -3,19 +3,39 @@ import Combine
 import Photos
 import XCTest
 import Yosemite
+import Networking
 
 final class ProductImageUploaderTests: XCTestCase {
     private let siteID: Int64 = 134
     private let productID = ProductOrVariationID.product(id: 606)
     private var errorsSubscription: AnyCancellable?
     private var assetUploadSubscription: AnyCancellable?
     private var activeUploadsSubscription: AnyCancellable?
+    private var mockFeatureFlagService: MockFeatureFlagService!
+    private var storage: ProductImageStatusStorage!
+
+    override func setUp() {
+        super.setUp()
+        mockFeatureFlagService = MockFeatureFlagService()
+        UserDefaults.standard.removeObject(forKey: "savedProductUploadImageStatuses")
```

Comment:
IMO it's better to inject the storage into the `ProductImageUploader` and test that the statuses are persisted correctly. We should avoid using `UserDefaults.standard` and use mock instances in unit tests.

Reply:
Yes, this is definitely a better approach. I applied the changes here (and fixed also a tests that was failing for me locally when running all the tests together): 2a7d9e5614b1ceb24edb0eb3cd3b58605d5e58d6